### ISSUE TITLE
Work out how to display executive positions when known

### ISSUE THREAD

**Issue by tmtmtmtm** [everypolitician/viewer-sinatra#13297]
# Problem

In many countries we now have in the source data information about who holds key positions: so for the UK we know who is Prime Minister, Chancellor of the Exchequer, Home Secretary etc. We don't currently show that information anywhere. We should.

# Proposal

We have a few possible options. 

1. We simply add this information to a person's card on the Term pages — either as a new 'view' of that, or in the header. 

2. We 'hoist' the key people (e.g. Cabinet-level members, in countries where that is applicable) to the top of the list.

3. We add a whole new set of pages for this.

My suspicion that that although we want to evolve towards #3 over time there's a quick win from a combination of #1 and #2 whilst we get there (which probably requires page-per-person first)

# Acceptance Criteria

This ticket is currently only for working out what adding these could/should look like. Implementation can be another ticket. So we're really just looking for a design that will enable people to see at a glance all the people listed on http://www.parliament.uk/mps-lords-and-offices/government-and-opposition1/her-majestys-government/ when looking at http://everypolitician.org/uk/commons/term-table/56.html

# Not Required

In an ideal world, this will _also_ make it obvious that, say, David Cameron _was_ the Prime Minister during the same term (etc), but that isn't necessarily necessary for a first version. (We'll want to find a way of displaying things like that fairly quickly though, so it would be good if this version wasn't a dead end that made it much harder to get to there).

This will only need to work for countries where the Cabinet members are also part of the legislature. Handling countries where there's a separation of powers that precludes that can come later.

# Prerequisites

n/a

# Related Issues

n/a

# Due Date

n/a

# Design?

Yes — this is _entirely_ about design at this stage.

# Text?

Ideally what's going on will be so self-evident that we don't need any explanatory copy, but let's see :)

# Bloggable?

Once implemented across multiple countries this is probably worth shouting about.

**Comment by zarino** [comment]
### Notes from conversation with Tony:

* This is about displaying a politician’s membership of the *executive branch* of the country (rather than—or in addition to—the *legislative branch*)
* We don't have this info for many countries yet, but will get more soon.
* The UK is a bit weird in that you can be *both* an MP for a constituency (a legislative branch position) *and* a member of the Cabinet (an executive branch position).
* In most countries, you *can’t* be a member of the executive *and* a member of the legislature at the same time. (This is why Barrack Obama doesn't appear on [EveryPolitician USA](http://everypolitician.org/united-states-of-america/)).
* It’s likely we’ll handle this for most countries as *just another House* to be displayed, with its own set of historic term_table pages (so, in the USA, [there would be three sections on this page]((http://everypolitician.org/united-states-of-america/): The House of Representatives, the Senate, and the Executive).
* Seems to me a complete no-brainer that we should display a politician’s membership of the executive just as we currently display legislative memberships.
   * Since most countries don't let you hold both at the same time, this will be totally natural on the Executive term_table pages for those countries, where Barrack Obama might look like:
   ![screen shot 2016-07-22 at 11 39 27](https://cloud.githubusercontent.com/assets/739624/17054791/fcbb7dd2-5000-11e6-913f-549fd48004d8.png)
   * Whereas, in countries like the UK, where you can hold both positions at once, the existing styling would just expand to accommodate:
   ![screen shot 2016-07-22 at 11 41 43](https://cloud.githubusercontent.com/assets/739624/17054850/4cad0bda-5001-11e6-86cc-c3e4ebbaadee.png)

### Questions left to be answered:

* On the pages for houses like the UK House of Commons, that allow members to *also* hold positions in the executive branch (like the UK Cabinet), do we differentiate these "special" members?
   * I was initially sceptical of how *special* these people should be made to appear. The page is about the legislative members, after all – the fact that a politician might also be Cabinet Minister for Silly Walks doesn't seem relevant enough to separate them out from the others.
   * Tony points out that it could be useful when you're navigating historical terms – showing the Prime Minister, for instance, would help you fairly quickly grasp the historical location of the presented term – even if it’s a position (Prime Minister) that isn’t technically part of the chosen house (The House of Commons).
* How do we make it clear to people visiting a term_table page for a legislative house (eg: [114th Congress of USA House of Representatives](http://everypolitician.org/united-states-of-america/house/term-table/114.html)) that there’s *another* page they might be interested in, which shows the US Executive / Cabinet, including famous faces like Barrack Obama and Joe Biden?

**Comment by tmtmtmtm** [comment]
> in countries like the UK, where you can hold both positions at once, the existing styling would just expand to accommodate

Is this only going to show the _current_ Cabinet position, or all the ones held during this term? e.g. should it show that Theresa May was Home Secretary before PM?

**Comment by tmtmtmtm** [comment]
I think it's going to be fairly common for people to have had 2 or 3 positions within a single term, but I suspect it drops off very quickly after that. But we may want to have a maximum number to display if even just to avoid massive imbalance in card sizes when there's someone who's been shuffled around a lot.

**Comment by zarino** [comment]
> But we may want to have a maximum number to display

My hypothesis would be that people will most likely care about either the politician’s current position (solves the "Where is the Prime Minister?" use-case) or, if the politician is not in a position any more, their most recent position (solves the "Where is the ex-Prime Minister?" use-case).

So maybe we display just the current or most recently held position, and then include a "And more…" link that just toggles the rest, for people who really want to dig down. (The "And more…" button also acts as a fairly clear signal to data geeks that there is more data under the surface.)

**Comment by tmtmtmtm** [comment]
Yep, I like that. (Though I suspect there's going to be some interesting edge cases with search: we'd presumably want Theresa May to be visible when someone searches for "Home Secretary", and for it to be obvious _why_ she's visible)

**Comment by chrismytton** [comment]
This is part of the wider issue of getting executive data onto the site, which is being tracked in https://github.com/everypolitician/viewer-sinatra/issues/24.

**Comment by chrismytton** [comment]
See https://github.com/everypolitician/viewer-sinatra/issues/24#issuecomment-280620076 for a summary of where we're at with the prototype for this.
